src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/Interchange.java

```diff
@@ -1,44 +1,63 @@
 package uk.nhs.digital.nhsconnect.nhais.model.edifact;
 
 import com.google.common.collect.ImmutableList;
+import lombok.Builder;
 import lombok.Getter;
+import lombok.Setter;
+import lombok.Singular;
 
 import java.util.List;
 import java.util.Optional;
 import java.util.stream.Collectors;
 
 @Getter
+@Setter
+@Builder
 public class Interchange {
 
     // TODO: stub for the internal representation of an edifact interchange
+    @Singular
     private final List<Segment> segments;
-
-    public Interchange(List<Segment> segments) {
-        this.segments = ImmutableList.copyOf(segments);
-    }
-
-    @Getter(lazy=true)
+    @Getter(lazy = true)
     private final InterchangeHeader interchangeHeader = findSegment(InterchangeHeader.class).orElseThrow();
-    @Getter(lazy=true)
+    @Getter(lazy = true)
     private final MessageHeader messageHeader = findSegment(MessageHeader.class).orElseThrow();
-    @Getter(lazy=true)
+    @Getter(lazy = true)
+    private final BeginningOfMessage beginningOfMessage = findSegment(BeginningOfMessage.class).orElseThrow();
+    @Getter(lazy = true)
     private final DateTimePeriod dateTimePeriod = findSegment(DateTimePeriod.class).orElseThrow();
-    @Getter(lazy=true)
+    @Getter(lazy = true)
     private final ReferenceTransactionNumber referenceTransactionNumber = findSegment(ReferenceTransactionNumber.class).orElseThrow();
-    @Getter(lazy=true)
+    @Getter(lazy = true)
     private final ReferenceTransactionType referenceTransactionType = findSegment(ReferenceTransactionType.class).orElseThrow();
-    @Getter(lazy=true)
+    @Getter(lazy = true)
     private final List<ReferenceMessageRecep> referenceMessageReceps = findMultipleSegments(ReferenceMessageRecep.class);
+    @Getter(lazy = true)
+    private final List<MessageTrailer> messageTrailer = findMultipleSegments(MessageTrailer.class);
```

Comment:
List is a work-around for multi-message for recep.
To be decided during https://gpitbjss.atlassian.net/browse/NIAD-224